src/views/VenuesListWithFilter/index.jsx

```diff
@@ -0,0 +1,74 @@
+import React, { useState, useEffect } from 'react'
+import CircularProgress from '@mui/material/CircularProgress';
+import styles from './styles.module.css'
+import { VenuesList } from './VenuesList'
+import { Pagination } from '@mui/material'
+import { FilterList } from './FilterList'
+
+const venuesUrl = 'http://localhost:3000/venues'
+
+export function VenuesListWithFilter () {
+  const [venues, setVenues] = useState(null)
+  const [venuesCache, setVenuesCache] = useState({})
+  const [currentPage, setCurrentPage] = useState(1)
+  const [amountOfVenues, setAmountOfVenues] = useState(null)
+  const [venuesPerPage, setVenuesPerPage] = useState(18)
+  const [currencyExchange, setCurrencyExchange] = useState(null)
+
+  useEffect(() => {
+    if (currentPage in venuesCache) {
+      setVenues(venuesCache[currentPage])
+      return
+    }
+    async function fetchVenues () {
+      const response = await fetch(`${venuesUrl}?_page=${currentPage}&_limit=${venuesPerPage}`)
+      const venuesArray = await response.json()
+      setAmountOfVenues(response.headers.get('X-Total-Count'))
+      setVenues(venuesArray)
+      setVenuesCache(({ ...venuesCache, [currentPage]: venuesArray }))
+    }
+    fetchVenues()
+      .catch(console.error)
+  }, [currentPage, venuesPerPage])
+
+  useEffect(() => {
+    fetch('https://api.apilayer.com/exchangerates_data/convert?to=PLN&from=EUR&amount=1',
+      {
+        headers: {
+          apikey: 'DtbpehV28fDVm4s3sLtJl3vWCAcH8cZU'
+        }
+      })
+      .then((res) => res.json())
+      .then(setCurrencyExchange)
+  }, [])
+
+  const handlePageChanging = (event, pageNumber) => setCurrentPage(pageNumber)
+  const handleAmountOfVenuesOnPageChange = (event) => {
+    setVenuesPerPage(event.target.value)
+    setVenuesCache({})
+  }
+
+  return (
+    <div className={styles.filtersAndVenuesWrap}>
+      <FilterList />
+      { venues && currencyExchange
```

Comment:
using && along with conditional statement is not very readable. You can think of different solution, or introduce a well-named variable